webapp/src/components/common/Matrix/components/MatrixResize.tsx

```diff
@@ -0,0 +1,113 @@
+/**
+ * Copyright (c) 2025, RTE (https://www.rte-france.com)
+ *
+ * See AUTHORS.txt
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ *
+ * SPDX-License-Identifier: MPL-2.0
+ *
+ * This file is part of the Antares project.
+ */
+
+import { useState } from "react";
+import { Box, Button, TextField } from "@mui/material";
+import { useUpdateEffect } from "react-use";
+import Transform from "@mui/icons-material/Transform";
+import { resizeMatrix } from "../shared/utils";
+import { useTranslation } from "react-i18next";
+import useEnqueueErrorSnackbar from "@/hooks/useEnqueueErrorSnackbar";
+import { toError } from "@/utils/fnUtils";
+import { useMatrixContext } from "../context/MatrixContext";
+import { clamp } from "ramda";
+
+function MatrixResize() {
+  const { currentState, setState, isSubmitting: isMatrixSubmitting } = useMatrixContext();
+  const { t } = useTranslation();
+  const errorSnackBar = useEnqueueErrorSnackbar();
+  const initialColumnCount = currentState.data[0].length;
+  const [columnCount, setColumnCount] = useState(initialColumnCount);
+  const [isSubmitting, setIsSubmitting] = useState(false);
+
+  useUpdateEffect(() => {
+    setColumnCount(initialColumnCount);
+  }, [initialColumnCount]);
+
+  ////////////////////////////////////////////////////////////////
+  // Event Handlers
+  ////////////////////////////////////////////////////////////////
+
+  const handleMatrixResize = () => {
+    if (columnCount === initialColumnCount) {
+      return;
+    }
+
+    setIsSubmitting(true);
+
+    try {
+      const updatedMatrix = resizeMatrix({
```

Comment:
More secure:

```typescript
setState((prev) => ({
  ...prev,
  data: resizeMatrix({
    matrix: prev.data,
    newColumnCount: columnCount,
  }),
  updateCount: currentState.updateCount + 1,
});
```

Reply:
its not like the setState of React, so it doesn’t support the functional update form 
type signature: `const setState: (newPresent: DataState, checkpoint?: boolean | undefined) => void`

Reply:
I thought it was the React one

Reply:
I rename it to avoid confusing with React's setState